gef-ui/server/controllers/amendments/cover-end-date/get-cover-end-date.ts

```diff
@@ -0,0 +1,48 @@
+import { CustomExpressRequest } from '@ukef/dtfs2-common';
+import { Response } from 'express';
+import * as api from '../../../services/api';
+import { CoverEndDateViewModel } from '../../../types/view-models/amendments/cover-end-date-view-model';
+import { asLoggedInUserSession } from '../../../utils/express-session';
+import { userCanAmendFacility } from '../../../utils/facility-amendments.helper';
+
+export type GetCoverEndDateRequest = CustomExpressRequest<{
+  params: { dealId: string; facilityId: string; amendmentId: string };
+}>;
+
+/**
+ * Renders the new cover end date view based on the provided request parameters
+ * @param {GetCoverEndDateRequest} Request object
+ * @param {Response} Response used to render the cover end date view
+ * @returns {void} Renders the 'partials/amendments/cover-end-date.njk' template
+ * or an error if rendering fails
+ */
+export const getCoverEndDate = async (req: GetCoverEndDateRequest, res: Response) => {
+  try {
+    const { dealId, facilityId, amendmentId } = req.params;
+    const { userToken, user } = asLoggedInUserSession(req.session);
+
+    const deal = await api.getApplication({ dealId, userToken });
+    const { details: facility } = await api.getFacility({ facilityId, userToken });
+
+    if (!deal || !facility) {
+      console.error('Get cover end date failed, due to missing data %o', deal, facility);
```

Comment:
sorry, this should be `%o %o` since 2 objects

Reply:
or do we need to print deal and facility since they will be null/undefined? or is `console.error('Get cover end date failed, due to missing data);` enough

Reply:
Done